tests/losses/test_contrastive_loss.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+import torch
+
+from fl4health.losses.contrastive_loss import ContrastiveLoss
+
+DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+def test_contrastive_loss() -> None:
+    # Default temperature is 0.5
+    contrastive_loss = ContrastiveLoss(DEVICE)
+
+    global_features = torch.tensor([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
+    local_features = torch.tensor([[[1.0, 2.0], [2.0, 1.0]], [[2.0, 1.0], [1.0, -1.0]]])
+    previous_local_features = torch.tensor([[[1.0, 2.0], [2.0, 1.0]], [[1.0, 1.0], [1.0, 1.0]]])
+
+    loss_value = contrastive_loss(
+        features=local_features.reshape(len(local_features), -1),
+        positive_pairs=global_features.reshape(1, len(global_features), -1),
+        negative_pairs=previous_local_features.reshape(1, len(previous_local_features), -1),
+    )
+
```

Comment:
I would also prefer to check device of input tensors whether they are actually on same device as loss or not.

Reply:
Yeah that's a good call.

Reply:
Because our github machines are all cpu driven. It's hard to write an "automatic" test for this. However, I logged into an A40 on the cluster and ran the following script to verify that the tensors move properly.
```
import torch

from fl4health.losses.perfcl_loss import PerFclLoss

DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

perfcl_loss_function = PerFclLoss(DEVICE, global_feature_loss_temperature=0.5, local_feature_loss_temperature=0.5)

local_features = torch.tensor([[1, 1, 1], [1, 1, 1]]).float().to("cpu")
global_features = torch.tensor([[1, 1, 1], [1, 1, 1]]).float().to("cpu")
old_local_features = torch.tensor([[0, 0, 0], [0, 0, 0]]).float().to("cpu")
old_global_features = torch.tensor([[0, 0, 0], [0, 0, 0]]).float().to("cpu")
initial_global_features = torch.tensor([[1, 1, 1], [1, 1, 1]]).float().to("cpu")

global_feature_loss, local_feature_loss = perfcl_loss_function(
    local_features,
    old_local_features,
    global_features,
    old_global_features,
    initial_global_features,
)
```
and everything worked well!